cpp/src/fil/infer.cu

```diff
@@ -172,7 +172,8 @@ struct tree_aggregator_t {
     : tmp_storage(finalize_workspace) {}
 
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, float> single_tree_prediction, int tree) {
+    vec<NITEMS, float> single_tree_prediction, int tree, int num_rows) {
+    for (int i = num_rows; i < NITEMS; ++i) single_tree_prediction[i] = 0.0f;
```

Comment:
I don't think the compiler will be able to unroll this loop.

Besides, I think it would be easier to change the loop in `infer_one_tree()`, this will also avoid a lot of changes:

```
vec<NITEMS, output_type> out;  // initialized to 0
#pragma unroll 
for (int j = 0; j < NITEMS; ++j) {
  if (j < n_rows) out[j] = tree[curr[j]].template output<output_type>();
}
```